lighthouse-core/test/gather/driver-test.js

```diff
@@ -240,6 +240,24 @@ describe('Browser Driver', () => {
     });
   });
 
+  it('.sendCommand timesout when commands take too long', async () => {
+    const driver = new Driver(connection);
+    driver._innerSendCommand = () => {
+      return new Promise(resolve => setTimeout(resolve, 15));
+    };
+
+    driver.setNextProtocolTimeout(25);
+    await driver.sendCommand('Page.enable');
+
+    driver.setNextProtocolTimeout(5);
+    try {
+      await driver.sendCommand('Page.disable');
+      assert.fail('expected driver.sendCommand to timeout');
+    } catch (err) {
+      // :)
```

Comment:
well here we want to assert that we didn't fail with the `assert.fail` you added :)

let's make sure that the message matches roughly what we want it to `.includes('PROTOCOL_TIMEOUT')` maybe?

Reply:
looks correct to me - if this command times out, the next statement in the `try` block should never be reached.

added check for err.code

Reply:
> looks correct to me - if this command times out, the next statement in the try block should never be reached.

If the command times out, it would've thrown. If it doesn't time out as expected, it would've also thrown on the following line with the fail. Both paths led to an empty catch block. New error code assertion looks perfect though!! 👌

Reply:
Ah, of course. I convinced myself that assert.fail didn't throw an error.. I was applying a pattern that works for other testing frameworks. But totally doesn't work here :p